app/mattermost.js

```diff
@@ -269,8 +269,11 @@ export default class Mattermost {
         setServerVersion('')(dispatch, getState);
     };
 
-    restartApp = () => {
+    restartApp = async () => {
         Navigation.dismissModal({animationType: 'none'});
+
+        const {dispatch, getState} = store;
+        await loadConfigAndLicense()(dispatch, getState);
```

Comment:
In order to retrieve custom emoji's after we reset the cache we need the config. Well since we're wiping the cache `getConfig(state).EnableCustomEmoji === 'true'` will always be false since config is just a blank object. From what I found the entire config was not being reloaded after the app was reset.